File: programming 1/lab 6 kotlin/src/main/java/server/commands/InfoCommand.java
package server.commands;

import common.communication.Response;
import server.managers.CollectionManager;

/**
 * Команда для вывода информации о коллекции.
 */
public class InfoCommand extends Command {
    private final CollectionManager cm;
    public InfoCommand(CollectionManager cm) {
        super("info", "информация о коллекции");
        this.cm = cm;
    }

    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    @Override
    public Response execute(String arg) {
        if (arg != null) throw new IllegalArgumentException(String.format("Команда %s не принимает аргументы\n", getName()));
        return cm.info();
    }
}

File: programming 1/lab 6/kotlin/lab 6/src/main/java/server/commands/InfoCommand.kt
package server.commands

import common.communication.Response
import server.managers.CollectionManager

/**
 * Команда для вывода информации о коллекции.
 */
class InfoCommand(private val cm: CollectionManager) : Command("info", "информация о коллекции") {
    /**
     * Выполняет команду без аргументов.
     * @param arg аргумент команды (в данной команде не используется).
     */
    override fun execute(arg: String?): Response {
        require(arg == null) { "Команда $name не принимает аргументы" }
        return cm.info()
    }
}
